README.md

```diff
@@ -17,24 +17,40 @@ For more information, visit the CrownLabs website ([https://crownlabs.polito.it]
 
-* **Frontend**, which is responsible to access Kubernetes API, guiding the user to creation of VMs.
-* **Laboratory Operator**, which reacts to LabInstances creation by creating the Kubernetes objects to launch
-the laboratory.
-
-A high-level representation of the main architectural building blocks composing CrownLabs is given by the following figure. Please notice that, for the sake of clarity, the figure depicts the elements essential for the provision of the actual service (i.e. remote computing labs), while leaving out with those more low-level or associated with the cluster operation (e.g monitoring).
+A high-level representation of the main architectural building blocks composing CrownLabs is given by the following figure.
+Please notice that, for the sake of clarity, the figure depicts the elements essential for the provision of the actual service (i.e. remote computing labs), while leaving out with those more low-level or associated with the cluster operation (e.g monitoring).
 
 ![CrownLabs High-Level Architecture](documentation/architecture.svg)
 
+## Backend Business Logic
+
+The backend business logic providing the different CrownLabs functionalities is implemented by custom Kubernetes operators, while the data model is defined by means of CRDs.
+Specifically, the main backend components are:
+
+* the Instance Operator, which implements the logic to spawn new environments starting from predefined templates;
+* the Tenant Operator, which automates the management of CrownLabs users (i.e. tenants) and groups (i.e. workpaces);
+* the Bastion Operator, which configures an SSH bastion to provide command-line access to the environments.
+
+Furthermore, some additional components are leveraged to simplify and automate companion tasks, such as listing the available images and deleting stale environments.
+
+For more information regarding the CrownLabs backend, as well as for the deployment and configuration instructions, please refer to the corresponding [README](./operators/README.md).
+
+## Frontend Dashboard
+
+The frontend dashboard is the component responsible for providing access to the CrownLabs custom resources through an easy to use graphical interface.
+It allows final users to spawn new environments, as well as to define new templates.
+Additionally, privileged users can create, update and delete both tenant and workspace resources, effectively managing the permissions granted to access the system.
 
-## External libraries
+The CrownLabs dashboard builds on top of [LiqoDash](https://github.com/liqotech/dashboard), a dynamic and general purpose dashboard designed to display and interact with Kubernetes resources.
+Specifically, it can be thought as a graphical kubectl, which directly interacts with the Kubernetes API server and provides access to both native and custom resources (i.e. those defined through CRDs).
```

Comment:
```suggestion
Specifically, it can be thought as a graphical `kubectl`, which directly interacts with the Kubernetes API server and provides access to both native and custom resources (i.e. those defined through CRDs).
```